src/mod/endpoints/mod_sofia/sofia_presence.c

```diff
@@ -2987,12 +2987,27 @@ static int sofia_presence_sub_callback(void *pArg, int argc, char **argv, char *
 
 			if (!strcasecmp(astate, "early") || !strcasecmp(astate, "confirmed")) {
 
+				char *cmd;
+				char *callee_id_name = NULL;
+				switch_stream_handle_t user_data_stream = { 0 };
+
 				clean_to_user = switch_mprintf("%s", sub_to_user ? sub_to_user : to_user);
 				clean_from_user = switch_mprintf("%s", from_id ? from_id : from_user);
 
+				SWITCH_STANDARD_STREAM(user_data_stream);
+				cmd = switch_mprintf("%s@%s var effective_caller_id_name", clean_to_user, host);
+				if (switch_api_execute("user_data", cmd, NULL, &user_data_stream) == SWITCH_STATUS_SUCCESS) {
```

Comment:
I have concerns about the need of calling the `user_data` API. Shouldn't we have all the information in the `helper->event` ?
`switch_event_get_header(helper->event, ...)`

Reply:
I don't have good knowledge of that object. But I would wonder why the effective_caller_id_name would be included. We are handling an inbound call and need to retrieve userdata for the notifying extension. Up until this point in the call flow this data is irrelevant.

Reply:
I enabled the prescence debug that does a DUMP_EVENT(helper->event) and can confirm that the information I'm looking for is not available without calling the API.